src/libre/ac.h

```diff
@@ -0,0 +1,50 @@
+#ifndef AC_H
+#define AC_H
+
+#define MODULE_NAME AC
+
+#include <stdio.h>
+
+struct fsm;
+struct fsm_state;
+struct fsm_options;
+
+struct trie_state {
+	struct trie_state *children[256];
+	struct trie_state *fail;
+	struct fsm_state *st;
+	unsigned int index;
+	unsigned int output:1;
+};
```

Comment:
@Caslyn pointed out a radix tree should better suit the spread for this data (assuming strings are fairly randomish, with a relatively narrow alphabet), because the shared initial prefix is short relative to the lengths of the strings.

We could replace this with a stock radix tree, but we'd still need the `.fail` state. So either we roll our own for that, or we'd import and modify an existing one.